### PR TITLE
fix: account deletion logic to keep boost funds into consideration

### PR DESCRIPTION
**Pull request by marcellorigotti**
# Pull Request

Closes: PRO-1337

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have updated documentation where appropriate.

## Summary

Check boost balances of LP before allowing de-registration

#### Non-Breaking changes

### CODE REVIEW

File: state-chain/pallets/cf-lp/src/lib.rs [hunk 1]
```diff
@@ -306,6 +314,12 @@ pub mod pallet {
 				Self::free_balances(&account_id)?.iter().all(|(_, amount)| *amount == 0),
 				Error::<T>::FundsRemaining
 			);
+			ensure!(
```
Comment (kylezs) [inline state-chain/pallets/cf-lp/src/lib.rs:317]: we could add a test for this with a slightly smarter mock

File: state-chain/pallets/cf-lp/src/lib.rs [hunk 2]
```diff
@@ -294,6 +301,7 @@ pub mod pallet {
 		pub fn deregister_lp_account(who: OriginFor<T>) -> DispatchResult {
 			const STABLE_ASSET: Asset = Asset::Usdc;
 			let account_id = T::AccountRoleRegistry::ensure_liquidity_provider(who)?;
+			T::PoolApi::sweep(&account_id)?;
```
Comment (kylezs) [inline state-chain/pallets/cf-lp/src/lib.rs:304]: why do we sweep here? I mean, i understand that we don't want to deregister if they have orders, but why not make them do this explicitly?

Comment (marcellorigotti) [reply state-chain/pallets/cf-lp/src/lib.rs:304]: wdym? as far as I understood sweep doesn't close any order, it collects just the fees.
If they have open orders the extrinsic will fail on the next check (when the other PR updating `open_orders_count` to include zero sized orders will be merged)

File: state-chain/pallets/cf-lp/src/lib.rs (hunk 1 above)
Comment (marcellorigotti) [reply state-chain/pallets/cf-lp/src/lib.rs:317]: yeah sure! what did you have in mind?

Comment (kylezs) [reply state-chain/pallets/cf-lp/src/lib.rs:317]: Just that this error is thrown when there are boost funds basically

**Review by kylezs — APPROVED** [review]

**Pull request merged**